core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -677,9 +684,54 @@ public PlanWithProperties visitFilter(FilterNode node, PreferredProperties prefe
         @Override
         public PlanWithProperties visitTableScan(TableScanNode node, PreferredProperties preferredProperties)
         {
+            // attempt tp push preferred partitioning into the table scan
+            var partitioningProperties = preferredProperties.getGlobalProperties().flatMap(PreferredProperties.Global::getPartitioningProperties);
```

Comment:
Are we using `var` in the code base now ? I thought we were only using it for ignored variables in `try` scope